### PR TITLE
We are pedants are we not

### PR DESCRIPTION
**Pull request by DavidSchinazi**
https://github.com/martinthomson/i-d-template/pull/452#issuecomment-2418322109

### CODE REVIEW

File: upload.mk [hunk 1]
```diff
@@ -32,7 +32,7 @@ endif
 	  file="$$(git ls-files "$${1%-[0-9][0-9]}.*")"; \
 	  for last in $$(git log --follow --name-only --format=format: -- "$${file%-[0-9][0-9]}" | \
 		sed -e '/^$$/d' | grep -v draft-todo-yourname-protocol | cut -f 2 | uniq | tail +2); do \
-	    if [ -n "$$(git tag -l "$${last%.*}-*")" ]; then \
+	    if [ -n "$$(git tag -l | grep -E "$${last%.*}-[0-9]{2}")" ]; then \
```
Comment (martinthomson) [inline upload.mk:35]:
```suggestion
	    if [ -n "$$(git tag -l "$${last%.*}-[0-9][0-9]")" ]; then \
```

**Review by martinthomson — CHANGES_REQUESTED** [review]
git is happy to use globs, so this is probably cleaner.

File: upload.mk (hunk 1 above)
Comment (DavidSchinazi) [inline upload.mk:35]:
```suggestion
	    if [ -n "$$(git tag -l "$${last%.*}-*")" | grep -E "$${last%.*}-[0-9]{2}")" ]; then \
```

File: upload.mk [hunk 2]
```diff
@@ -32,7 +32,7 @@ endif
 	  file="$$(git ls-files "$${1%-[0-9][0-9]}.*")"; \
 	  for last in $$(git log --follow --name-only --format=format: -- "$${file%-[0-9][0-9]}" | \
 		sed -e '/^$$/d' | grep -v draft-todo-yourname-protocol | cut -f 2 | uniq | tail +2); do \
-	    if [ -n "$$(git tag -l "$${last%.*}-*")" ]; then \
+	    if [ -n "$$(git tag -l "$${last%.*}-*")" | grep -E "$${last%.*}-[0-9]{2}")" ]; then \
```
Comment (DavidSchinazi) [inline upload.mk:35]:
```suggestion
	    if [ -n "$$(git tag -l | grep -E "^$${last%.*}-[0-9]{2}$$")" ]; then \
```

File: upload.mk [hunk 3]
```diff
@@ -32,7 +32,7 @@ endif
 	  file="$$(git ls-files "$${1%-[0-9][0-9]}.*")"; \
 	  for last in $$(git log --follow --name-only --format=format: -- "$${file%-[0-9][0-9]}" | \
 		sed -e '/^$$/d' | grep -v draft-todo-yourname-protocol | cut -f 2 | uniq | tail +2); do \
-	    if [ -n "$$(git tag -l "$${last%.*}-*")" ]; then \
+	    if [ -n "$$(git tag -l | grep -E "^$${last%.*}-[0-9]{2}$$")" ]; then \
```
Comment (martinthomson) [inline upload.mk:35]:
```suggestion
	    if [ -n "$$(git tag -l "$${last%.*}-[0-9][0-9]")" ]; then \
```

You might have missed the simpler fix...

Comment (DavidSchinazi) [reply upload.mk:35]: I thought about it, but that matches the tag `aaaaaaa-draft-stuff-foobar-03`. Not sure if that matters

Comment (martinthomson) [reply upload.mk:35]: I don't think it does.  If `git tag -l a` doesn't match "ab", then this won't match `aaaaa-draft-whatever-09`.

Comment (DavidSchinazi) [reply upload.mk:35]: Oh. Duh

**Pull request merged**